whitepapers/0007_pyth_over_wormhole.md

```diff
@@ -0,0 +1,208 @@
+| pyth2wormhole                     | on-chain Solana program in Rust                       | When called, retrieves Pyth data for the specified asset and posts it to Wormhole's solana program.                                                                                                                                                                                                                                                |
+| pyth2wormhole-client              | off-chain Rust program                                | When run, this off-chain program calls pyth2wormhole to forward the current price data of an asset to Wormhole.                                                                                                                                                                                                                                    |
+| Pyth relayer software (name TBD?) | off-chain program                                     | When active, this program acts as a Wormhole relayer. A relayer listens to peer-to-peer Wormhole guardian messaging for data coming from the supported chains. When our relayer detects relevant Pyth price data, it will complete the bridging effort by relaying the price data to a specific smart contract on the requested target blockchain. |
+| Target chain smart contracts      | on-chain smart contract for each desired target chain | When called, each target chain smart contract will verify the provided Pyth data and make it available to target chain users.                                                                                                                                                                                                                      |
+
+### API / database schema
+
+### A note on Pyth data
+Pyth's storage model maintains a linked-list-style collection of
+accounts that describe multiple assets. Each asset contains its own
+list of price data of different types, which themselves contain prices
+from various quoter accounts that name concrete numbers as current
+price value. The individual source values are also available as a
+single, aggregated number from the upper-level asset price
+description.
+
+In the current system metadata from arbitrary Pyth feeds can not be
+trusted and anyone can create new feeds for popular symbols. Projects
+that build products on top of Pyth need to maintain a list of the
+product_ids (Solana account keys) that they trust.  As long as this is
+the case we can also rely on projects to hardcode the relevant
+metadata that they require in their business logic and focus this
+initial release on just attesting pricefeeds.
```

Comment:
Is this the case? I thought the pyth foundation or someone had to create new product metadata and permission it.

Reply:
Kinda, def need to reword this.